### DISCUSSION TITLE
Websocket Primer - Beginner

### DISCUSSION THREAD

**Discussion by aifrog** [bbernhard/signal-cli-rest-api#461] [Q&A]
The API works for me using GET/POST.

I now switched to json-rpc and try to open a Websocket using a tool:

![Screenshot 2023-12-16 134529](https://github.com/bbernhard/signal-cli-rest-api/assets/51208844/b8f9788a-1ca9-4030-8200-262e4f5ecab1)

However, I get an Error. Anybody has a hint? How can I test the Websocket from a linux shell?

Best, Hans-Christian

**Comment by bbernhard** [comment]
What's the actual error?

**Reply by aifrog** [reply]
Sorry, war my error. Replacing http with ws did the trick. Thx.

**Comment by aifrog** [comment]
I am now able to successfully **receive** and process the JSON object of a signal message via web socket. Great!

I am also able to **send** some plain text through the web socket. But nothing happens in Signal - probably this has to be a specific json format as well. How should I build this JSON object to make it work?

Thx, H-C

**Comment by bbernhard** [comment]
see https://bbernhard.github.io/signal-cli-rest-api/ for the swagger documentation

and https://github.com/bbernhard/signal-cli-rest-api/blob/master/doc/EXAMPLES.md for concrete examples.

**Comment by aifrog** [comment]
I saw that and I am able to send via Curl (not web socket). The string

`{"message":"This` is another test message","number":"+49XXXXXXXX","recipients":["group.NXXXXXXXXXXXXXXXXX"]}`

works well via curl. Can I send this right through the web socket as well?

**Reply by bbernhard** [reply]
Only the `receive` endpoint is implemented via websockets. Sending a message is always done via a HTTP POST.

**Reply by aifrog** [reply]
Thx.
